src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -414,23 +438,23 @@ private bool TrySteal(out IThreadPoolWorkItem obj, ref bool missedSteal, int mil
         internal class QueueSegment
         {
             // Holds a segment of the queue.  Enqueues/Dequeues start at element 0, and work their way up.
-            internal readonly IThreadPoolWorkItem[] nodes;
+            internal readonly PaddedWorkItem[] nodes;
```

Comment:
Additionally, now when accessing these we need to touch one cache line per workitem, vs. one line per 8 work items in the current code.  I'd expect this to result in quite a lot more cache churn, and thus much slower execution, in many scenarios.